src/modules/userMethods.ts

```diff
@@ -460,4 +461,124 @@ export const userMethods = {
             active: user.active,
         };
     },
+    createUserRequest: async (data: NonNullableFields<CreateUser, 'groupId' | 'supervisorId'>) => {
+        const name = trimAndJoin([data.surname, data.firstName, data.middleName]);
+
+        const supervisor = await prisma.user.findUniqueOrThrow({ where: { id: data.supervisorId ?? undefined } });
+
+        if (!supervisor.login) {
+            throw new TRPCError({ code: 'INTERNAL_SERVER_ERROR', message: 'Supervisor has no login' });
+        }
+
+        if (!data.groupId) {
+            throw new TRPCError({ code: 'INTERNAL_SERVER_ERROR', message: 'Group is required' });
+        }
+
+        return prisma.userCreationRequest.create({
+            data: {
+                name,
+                supervisorLogin: supervisor.login,
+                email: data.email,
+                login: data.login,
+                organizationUnitId: data.organizationUnitId,
+                groupId: data.groupId,
+                createExternalAccount: data.createExternalAccount,
+                services: {
+                    toJSON: () => [
+                        {
+                            serviceName: 'Phone',
+                            serviceId: data.phone,
+                        },
+                        {
+                            serviceName: 'Accounting system',
+                            serviceId: data.accountingId,
+                        },
+                    ],
+                },
+            },
+        });
+    },
+
+    getUsersRequests: async <
+        T extends UserCreationRequest & {
+            status: 'Approved' | 'Denied' | null;
+            supervisor: User;
+            organization: OrganizationUnit;
+            group: Group;
+            services: Record<'serviceName' | 'serviceId', string>[] | null;
+        },
+    >(): Promise<T[]> => {
+        const requests = await prisma.userCreationRequest.findMany({
+            where: {
+                status: null,
+            },
+            include: {
+                group: true,
+                organization: true,
+                supervisor: true,
+            },
+        });
+
+        return requests as T[];
+    },
+
+    declineUserRequest: async ({ id, comment }: { id: string; comment?: string }) => {
+        return prisma.userCreationRequest.update({
+            where: { id },
+            data: { status: 'Denied', comment },
+        });
+    },
+
+    acceptUserRequest: async ({ id, comment }: { id: string; comment?: string }) => {
+        const acceptedRequest = await prisma.userCreationRequest.update({
+            where: { id },
+            data: { status: 'Approved', comment },
+            include: { supervisor: true },
+        });
+
+        const services = acceptedRequest.services as { serviceId: string; serviceName: string }[];
+
+        const checkUser = await userMethods.getByLogin(acceptedRequest.login);
```

Comment:
Check can be made before accepting request, otherwise we can get accepted request and an error at the same time